crates/matrix-sdk-indexeddb/src/crypto_store/mod.rs

```diff
@@ -827,13 +827,25 @@ impl_crypto_store! {
 
         let mut result = Vec::new();
         for _ in 0..limit {
-            result.push(self.deserialize_inbound_group_session(cursor.value())?);
+            result.push(cursor.value());
             if !cursor.continue_cursor()?.await? {
                 break;
             }
         }
 
         tx.await.into_result()?;
+
+        // deserialize and decrypt after the transaction is complete
+        let result = result.into_iter()
+            .filter_map(|v| match self.deserialize_inbound_group_session(v) {
+                Ok(session) => Some(session),
+                Err(e) => {
+                    warn!("Failed to deserialize inbound group session: {:?}", e);
```

Comment:
We should be able to `Display` the error, so this should work:

```suggestion
                    warn!("Failed to deserialize inbound group session: {e}");
```